### PR TITLE
[MOB-801] Move deviceInfo to the top level of in-app event payloads

### PR DESCRIPTION
**Pull request by vbabenkoru**

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 1]
```diff
@@ -1099,6 +1100,7 @@ public void trackInAppClick(IterableInAppMessage message, String clickedUrl, Ite
             requestJSON.put(IterableConstants.KEY_MESSAGE_ID, message.getMessageId());
             requestJSON.put(IterableConstants.ITERABLE_IN_APP_CLICKED_URL, clickedUrl);
             requestJSON.put(IterableConstants.KEY_MESSAGE_CONTEXT, getInAppMessageContext(message, clickLocation));
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: getInAppMessagContext also internally calls getDeviceInfoJson() -> which makes 

```
 "deviceInfo": {
        "deviceId": "cff3a019-51ad-4d15-919c-d3020617a7ec",
        "platform": "Android",
        "appPackageName": "com.iterable.androidsdktester"
      }
```

to be repeated in request. Is it done for transition purpose?

Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: Oh, I forgot to remove that, good catch! Thanks

Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: Fixed

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 2]
```diff
@@ -1144,6 +1146,9 @@ void trackInAppClose(IterableInAppMessage message, String clickedURL, IterableIn
             requestJSON.put(IterableConstants.ITERABLE_IN_APP_CLICKED_URL, clickedURL);
             requestJSON.put(IterableConstants.ITERABLE_IN_APP_CLOSE_ACTION, closeAction.toString());
             requestJSON.put(IterableConstants.KEY_MESSAGE_CONTEXT, getInAppMessageContext(message, clickLocation));
+            requestJSON.put(IterableConstants.KEY_DEVICE_INFO, getDeviceInfoJson());
+
+
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: Extra space 😬 ?

Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: Fixed!

**Pull request merged**